EquilateralTriangle.java

```diff
@@ -0,0 +1,41 @@
+import java.math.BigDecimal;
+
+/**
+ * Describes equilateral triangles.
+ */
+class EquilateralTriangle extends Triangle {
+  double repeatingSide;
+  private String type = new String("equilateral");
+  
+  /**
+   * EquilateralTriangle constructor.
+   *
+   * @throws IllegalArgumentException if the triangle can't exist.
+   */
+  public EquilateralTriangle(double[] sides) {
+    super(sides);
+    if (!exists(sides)) {
+      throw new IllegalArgumentException();
+    }
+    repeatingSide = sides[0];
+  }
+  
+  /**
+   * Determines the existence of an equilateral triangle with given sides.
+   *
+   * @return true if the triangle exists and is equilateral,
+   *         false if not.
+   */
+  public boolean exists(double[] sides) {
+    if (!super.exists(sides)) {
+      return false;
+    }  
+    BigDecimal sideA = new BigDecimal(sides[0]);
```

Comment:
avoid that cast from double to BigDecimal - let sides be BigDecimal from very beginning